client/app/leaderboard/page.tsx

```diff
@@ -36,12 +37,15 @@ export default function Page() {
 
     return (
         <div className="flex h-screen flex-col">
-            <div className="m-2">
-                <Button variant="ghost" asChild>
-                    <Link href="/">
-                        <ArrowLeft /> Back
-                    </Link>
-                </Button>
+            <div className="m-2 flex justify-between">
+                <div className="duration-250 relative -m-16 p-16 opacity-0 transition-opacity hover:opacity-100">
+                    <Button variant="ghost" asChild className="font-bold">
+                        <Link href="/">
+                            <ArrowLeft /> Back
+                        </Link>
+                    </Button>
+                </div>
+                <LeaderboardMenu />
             </div>
             <Leaderboard />
```

Comment:
I don't really like how this pushes the leaderboard timer down and adds a bunch of blank space on the top.  Can you push the timer to the top and keep the buttons where they are.
![screenshot](https://github.com/user-attachments/assets/4f232b43-b80d-4098-a54e-73265ef77226)